issue_processing/jira_issue_processing.py

```diff
@@ -403,108 +404,116 @@ def load_issue_via_api(issues, persons, url):
-                    history["date"] = format_time(change.created)
-                    histories.append(history)
-                    state_changes.append([history["date"], new_state])
-
-                # resolution_updated event gets created and added to the issue history
-                elif item.field == "resolution":
-                    if item.fromString is not None:
-                        old_resolution = item.fromString.lower()
-                    if item.toString is not None:
-                        new_resolution = item.toString.lower()
-                    history = dict()
-                    history["event"] = "resolution_updated"
-                    history["event_info_1"] = new_resolution
-                    history["event_info_2"] = old_resolution
-                    if hasattr(change, "author"):
-                        user = create_user(change.author.displayName, change.author.name, "")
-                    else:
-                        log.warn("No author for history: " + str(change.id) + " created at " + str(change.created))
-                        user = create_user("","","")
-                    history["author"] = merge_user_with_user_from_csv(user, persons)
-                    history["date"] = format_time(change.created)
-                    histories.append(history)
-                    resolution_changes.append([history["date"], new_resolution])
-
-                # assigned event gets created and added to the issue history
-                elif item.field == "assignee":
-                    history = dict()
-                    history["event"] = "assigned"
-                    user = create_user(change.author.displayName, change.author.name, "")
-                    history["author"] = merge_user_with_user_from_csv(user, persons)
-                    assignee = create_user(item.toString, item.to, "")
-                    assigned_user = merge_user_with_user_from_csv(assignee, persons)
-                    history["event_info_1"] = assigned_user["name"]
-                    history["event_info_2"] = assigned_user["email"]
-                    history["date"] = format_time(change.created)
-                    histories.append(history)
-
-                elif item.field == "Link":
-                    # add_link event gets created and added to the issue history
-                    if item.toString is not None:
+        # only consider history changes if we were able to extract the changelog for the current issue
+        if changelog is not None:
+
+            # history changes get visited in time order from oldest to newest
+            for change in changelog.histories:
+
+                # default values for state and resolution
+                old_state, new_state, old_resolution, new_resolution = "open", "open", "unresolved", "unresolved"
+
+                # all changes in the issue changelog are checked if they contain an useful information
```

Comment:
Since you are already fixing a/an errors above: It is a useful... not an useful...

Reply:
As comments and documentation were inconsistently formatted in general, I decided to streamline all the comments in this file. I hope you are fine with that :smile: